cloud_experts_tutorials/cloud-experts-deploying-application/cloud-experts-deploying-application-logging.adoc

```diff
@@ -0,0 +1,235 @@
+			"storedBytes": 0
+		},
+		{
+			"logGroupName": "rosa-fxxj9.infrastructure",
+			"creationTime": 1682098364399,
+			"metricFilterCount": 0,
+			"arn": "arn:aws:logs:us-east-2:511846242393:log-group:rosa-fxxj9.infrastructure:*",
+			"storedBytes": 0
+		}
+	]
+}
+----
+
+== Outputting the data to the streams and logs
+. Output a message to `stdout`.
+.. In the OSToy console, click *Home* and then click the message box for *Log Message (stdout)*.
+.. Write a message to output to the `stdout` stream, for example "All is well!".
+.. Click *Send Message*.
++
+image:cloud-experts-deploying-application-logging-ostoy-stdout.png[]
++
+. Output a message to `stderr`.
+.. Click the message box for *Log Message (stderr)*.
+.. Write a message to output to the `stderr` stream, for example "Oh no! Error!".
+.. Click *Send Message*.
++
+image:cloud-experts-deploying-application-logging-ostoy-stderr.png[]
+
+
+== Viewing the application logs by using the `oc` command
+. Enter the following command in the command line interface (CLI) to retrieve the name of your frontend pod:
++
+[source,terminal]
+----
+$ oc get pods -o name
+----
++
+.Example output
++
+[source,terminal]
+----
+pod/ostoy-frontend-679cb85695-5cn7x <1>
+pod/ostoy-microservice-86b4c6f559-p594d
+----
+<1> The pod name is `ostoy-frontend-679cb85695-5cn7x`.
++  
+. Run the following command to see both the `stdout` and `stderr` messages:
++
+[source,terminal]
+----
+$ oc logs ostoy-frontend-679cb85695-5cn7x
+----
++
+.Example output
++
+[source,terminal]
+----
+$ oc logs ostoy-frontend-679cb85695-5cn7x
+[...]
+ostoy-frontend-679cb85695-5cn7x: server starting on port 8080
+Redirecting to /home
+stdout: All is well!
+stderr: Oh no! Error!
+----
+
+== Viewing the logs with CloudWatch
+. Navigate to *CloudWatch* on the link:https://aws.amazon.com/free/?gclid=EAIaIQobChMIoq2X27fZhgMVaAutBh1Q2w1jEAAYASAAEgL5y_D_BwE&trk=7541ebd3-552d-4f98-9357-b542436aa66c&sc_channel=ps&ef_id=EAIaIQobChMIoq2X27fZhgMVaAutBh1Q2w1jEAAYASAAEgL5y_D_BwE:G:s&s_kwcid=AL!4422!3!651751058796!e!!g!!aws%20console!19852662149!145019243977&all-free-tier.sort-by=item.additionalFields.SortRank&all-free-tier.sort-order=asc&awsf.Free%20Tier%20Types=*all&awsf.Free%20Tier%20Categories=*all[AWS web console].
+. In the left menu, click *Logs* and then *Log groups* to see the different groups of logs. You should see 2 groups. One for `<cluster-name>-XXXXX-application` and one for `<cluster-name>-XXXXX-infrastructure`.
++
+image:cloud-experts-deploying-application-logging-cw.png[]
++
+. Click `<cluster-name>-XXXXX.application`.
+. Click the log stream for the frontend pod.
++
+image:cloud-experts-deploying-application-logging-logstream2.png[]
++
+. Filter for `stdout` and `stderr`.
+. Expand the row to show the messages you entered earlier and other pertinent information.
++
+image:cloud-experts-deploying-application-logging-stderr.png[]
++
+. Enter "microservice" in the search bar to see other messages in your logs.
```

Comment:
they need to go back to the log streams and select the microservice one first before this step.